test/dhtest.c

```diff
@@ -580,19 +491,23 @@ static int run_rfc5114_tests(void)
     BN_free(priv_key);
     OPENSSL_free(Z1);
     OPENSSL_free(Z2);
-
-    fprintf(stderr, "Initialisation error RFC5114 set %d\n", i + 1);
-    ERR_print_errors_fp(stderr);
+    TEST_error("Initialisation error RFC5114 set %d\n", i + 1);
```

Comment:
I'd take the latter option.  Any errors in the stack should be printed after the failed test.
I hadn't made it to making this change.

This means printing errors after any failed TEST_xxx conditional and TEST_error.  I'd be tempted to do it after TEST_info as well.